patroni/postgresql.py

```diff
@@ -92,18 +107,70 @@ def __init__(self, config):
             self.set_role('master' if self.is_leader() else 'replica')
             self._write_postgresql_conf()  # we are "joining" already running postgres
 
-    @staticmethod
-    def get_server_parameters(config):
-        return {p: v for p, v in (config.get('parameters') or {}).items() if p not in ('listen_addresses', 'port')}
+    @property
+    def use_slots(self):
+        return self._use_slots and self._major_version >= 9.4
+
+    def get_major_version(self):
+        if not self.data_directory_empty():
+            try:
+                with open(os.path.join(self._data_dir, 'PG_VERSION')) as f:
+                    return float(f.read())
+            except Exception:
+                logger.exception('Failed to read PG_VERSION from %s', self._data_dir)
+        return 0.0
+
+    def get_server_parameters(self, config):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/cyberdemn/patroni/file/2630856516/issues/source?bid=3127594&fileBranchId=3318079#l123)